### DISCUSSION TITLE
컨트롤러의 반환값 (feat. Thymeleaf)

### DISCUSSION THREAD

**Discussion by soeunnPark** [Kernel360/E2E2-CATCHLINE#28] [General]
타임리프 사용 시 컨트롤러의 반환타입이 `String`이고, templates의 `html` 파일을 반환해야 하는 걸로 알고 있습니다.
지금은 개발 중에 PostMan 등 툴에서 테스트하기 위해 컨트롤러에서 Json을 반환하고 있습니다.
개발 중에 html 파일을 같이 작성하며 만들어야 할까요?

**Comment by I-migi** [comment]
제 생각에는 GPT를 이용해서 간단하게 html 파일과 타임리프 파일을 만들어 동작이 제대로 되는지 확인 후 나중에 리팩토링을 하는 것도 좋을 것 같습니다.

**Reply by soeunnPark** [reply]
오 좋은 것 같아요 👍

**Comment by sangwonsheep** [comment]
현재 고민하시는 사항이 두 가지인 것 같아요.
1. @RestController 방식으로 기능 구현을 끝내고 Thymeleaf를 구현하기
2. @Controller 방식으로 기능 구현하면서 동시에 Thymeleaf도 구현하기
저는 개인적으로 했던 방식이 2번 방식이었어요. 화면을 그릴 때 필요한 데이터를 생각하느라 그랬던 것 같아요.
두 방식 중 어떤 방식이 좋은지는 모르겠지만 저는 2번 방식이 화면을 바로 볼 수 있어서 좋은 것 같아요.

**Reply by soeunnPark** [reply]
컨트롤러 어노테이션도 변경해야 하네요! 화면도 같이 그리면서 구현해봐야겠어요! 😄  감사합니다 ~!!
